diff -ruNa qmail-1.03.orig/qmail-smtpd.c qmail-1.03/qmail-smtpd.c
--- qmail-1.03.orig/qmail-smtpd.c	1998-06-15 18:53:16.000000000 +0800
+++ qmail-1.03/qmail-smtpd.c	2004-02-28 20:35:31.000000000 +0800
@@ -414,6 +414,7 @@
   if (chdir(auto_qmail) == -1) die_control();
   setup();
   if (ipme_init() != 1) die_ipme();
+  if (fd_copy(2,1) == -1) return err_pipe();
   smtp_greet("220 ");
   out(" ESMTP\r\n");
   if (commands(&ssin,&smtpcommands) == 0) die_read();
