diff -rua qmail-1.03.cocktail/conf-cc qmail-1.03/conf-cc
--- qmail-1.03.cocktail/conf-cc	2004-04-29 11:32:38.000000000 +0800
+++ qmail-1.03/conf-cc	2004-04-29 11:33:03.000000000 +0800
@@ -1,3 +1,3 @@
-cc -O2 -DTLS=20021228 -I/usr/local/ssl/include
+cc -O2 -DTLS=20021228
 
 This will be used to compile .c files.
diff -rua qmail-1.03.cocktail/conf-spawn qmail-1.03/conf-spawn
--- qmail-1.03.cocktail/conf-spawn	2004-04-29 11:32:38.000000000 +0800
+++ qmail-1.03/conf-spawn	2004-04-29 11:34:07.000000000 +0800
@@ -1,4 +1,4 @@
-1000
+509
 
 This is a silent concurrency limit. You can't set it above 255. On some
 systems you can't set it above 125. qmail will refuse to compile if the
diff -rua qmail-1.03.cocktail/qmail-queue.c qmail-1.03/qmail-queue.c
--- qmail-1.03.cocktail/qmail-queue.c	1998-06-15 18:53:16.000000000 +0800
+++ qmail-1.03/qmail-queue.c	2004-05-05 23:20:47.000000000 +0800
@@ -70,7 +70,7 @@
  unsigned int i;
  unsigned int len;
  len = 0;
- i = fmt_str(s,"Received: (qmail "); len += i; if (s) s += i;
+ i = fmt_str(s,"Received: (noSPAM "); len += i; if (s) s += i;
  i = fmt_ulong(s,mypid); len += i; if (s) s += i;
  i = fmt_str(s," invoked "); len += i; if (s) s += i;
  if (uid == auto_uida)
diff -rua qmail-1.03.cocktail/qmail-smtpd.c qmail-1.03/qmail-smtpd.c
--- qmail-1.03.cocktail/qmail-smtpd.c	2004-04-29 11:32:38.000000000 +0800
+++ qmail-1.03/qmail-smtpd.c	2004-05-05 20:37:56.000000000 +0800
@@ -963,6 +963,7 @@
   if (chdir(auto_qmail) == -1) die_control();
   setup();
   if (ipme_init() != 1) die_ipme();
+  if (fd_copy(2,1) == -1) return err_pipe();
   smtp_greet("220 ");
   out(" ESMTP\r\n");
   if (commands(&ssin,&smtpcommands) == 0) die_read();
diff -rua qmail-1.03.cocktail/received.c qmail-1.03/received.c
--- qmail-1.03.cocktail/received.c	2004-04-29 11:32:38.000000000 +0800
+++ qmail-1.03/received.c	2004-05-05 13:31:19.000000000 +0800
@@ -63,7 +63,7 @@
   safeput(qqt,remoteip);
   qmail_puts(qqt,")\n\tby ");
   safeput(qqt,local);
-  qmail_puts(qqt," (qmail 1.03 + ejcp v14) with ");
+  qmail_puts(qqt," (noSPAM V2) with ");
   qmail_puts(qqt,protocol);
   qmail_puts(qqt,";\n\t");
   datetime_tai(&dt,now());
diff -rua qmail-1.03.orig/qmail-qmqpd.c qmail-1.03/qmail-qmqpd.c
--- qmail-1.03.orig/qmail-qmqpd.c	1998-06-15 18:53:16.000000000 +0800
+++ qmail-1.03/qmail-qmqpd.c	2004-06-06 05:21:02.000000000 +0800
@@ -53,6 +53,8 @@
   }
 }
 
+char envelope_from[1000];
+
 void getcomma()
 {
   char ch;
@@ -78,7 +80,7 @@
   if (!local) local = env_get("TCPLOCALIP");
   if (!local) local = "unknown";
  
-  received(&qq,"QMQP",local,remoteip,remotehost,remoteinfo,(char *) 0);
+  received(&qq,"QMQP",local,remoteip,remotehost,remoteinfo,(char *) 0,envelope_from);
 }
 
 char buf[1000];
@@ -132,9 +134,11 @@
   }
   getcomma();
 
-  if (getbuf())
+  envelope_from[0]=0;
+  if (getbuf()){
     qmail_from(&qq,buf);
-  else {
+    str_copy(envelope_from,buf);
+  } else {
     qmail_from(&qq,"");
     qmail_fail(&qq);
     flagok = 0;
diff -rua qmail-1.03.orig/qmail-smtpd.c qmail-1.03/qmail-smtpd.c
--- qmail-1.03.orig/qmail-smtpd.c	1998-06-15 18:53:16.000000000 +0800
+++ qmail-1.03/qmail-smtpd.c	2004-06-06 04:21:41.000000000 +0800
@@ -378,7 +378,7 @@
   qp = qmail_qp(&qqt);
   out("354 go ahead\r\n");
  
-  received(&qqt,"SMTP",local,remoteip,remotehost,remoteinfo,fakehelo);
+  received(&qqt,"SMTP",local,remoteip,remotehost,remoteinfo,fakehelo,mailfrom.s);
   blast(&hops);
   hops = (hops >= MAXHOPS);
   if (hops) qmail_fail(&qqt);
diff -rua qmail-1.03.orig/received.c qmail-1.03/received.c
--- qmail-1.03.orig/received.c	1998-06-15 18:53:16.000000000 +0800
+++ qmail-1.03/received.c	2004-06-06 05:14:39.000000000 +0800
@@ -37,7 +37,7 @@
 /* "Received: from relay1.uu.net (HELO uunet.uu.net) (7@192.48.96.5)\n" */
 /* "  by silverton.berkeley.edu with SMTP; 26 Sep 1995 04:46:54 -0000\n" */
 
-void received(qqt,protocol,local,remoteip,remotehost,remoteinfo,helo)
+void received(qqt,protocol,local,remoteip,remotehost,remoteinfo,helo,envelope_from)
 struct qmail *qqt;
 char *protocol;
 char *local;
@@ -45,6 +45,7 @@
 char *remotehost;
 char *remoteinfo;
 char *helo;
+char *envelope_from;
 {
   struct datetime dt;
 
@@ -63,6 +64,9 @@
   safeput(qqt,remoteip);
   qmail_puts(qqt,")\n  by ");
   safeput(qqt,local);
+  qmail_puts(qqt," (envelope-from ");
+  safeput(qqt,envelope_from);
+  qmail_puts(qqt,")\n\t");
   qmail_puts(qqt," with ");
   qmail_puts(qqt,protocol);
   qmail_puts(qqt,"; ");
