--- Simple.orig.pm       2004-02-19 17:47:57.000000000 +0800
+++ Simple.pm   2004-02-19 18:08:52.000000000 +0800
@@ -973,9 +977,14 @@
   my $count = 0;
   if($self->{opt}->{keyattr}) {
     while(($key,$val) = each %$attr) {
-      if(defined($val)  and  UNIVERSAL::isa($val, 'ARRAY')) {
+     if(defined($val)  and  UNIVERSAL::isa($val, 'ARRAY')) {
         $attr->{$key} = $self->array_to_hash($key, $val);
+      }elsif ( $key =~ /[A-Z]/ ){
+       delete $attr->{$key};
+       $key = lc $key;
+       $attr->{$key} = $val;
       }
+ 
       $count++;
     }
   }
